src/flakeref/fr_type.rs

```diff
@@ -338,12 +260,7 @@ impl FlakeRefType {
                 *ref_or_rev = ref_or_rev_alt;
             }
             // TODO: #158
-            FlakeRefType::Mercurial { .. }
-            | FlakeRefType::Path { .. }
-            | FlakeRefType::Tarball { .. }
-            | FlakeRefType::File { .. }
-            | FlakeRefType::Git { .. }
-            | FlakeRefType::None => todo!(),
+            _ => return Err(NixUriError::InvalidType(self.to_string())),
```

Comment:
@a-kenji could you please confirm that this is what we want here and at the other `get` functions??

Reply:
I think this is OK, a clearer error would be, `NotSupportedByType`, or something along the lines of that?

Reply:
done: `rebase -i main` -> edited first commit: see first commit to see the changes. 

If happy, mark this as resolved and I'll do a final self-review before merge.